run_hs_metrics.sh

```diff
@@ -29,29 +32,36 @@ CONTAINER=/vulpes/ngi/containers/sarek/nfcore-sarek-2.7.img
 if [[ $# -eq 3 ]]
 then
   
-  # This is the functionality when the user runs the script with 3 parameters. BAM files will be located
+  # This is the functionality when the user runs the script with 3 parameters. BAM/CRAM files will be located
   # and sbatch jobs for collecting metrics dispatched
 
-  find "${ANALYSISDIR}/results/Preprocessing" -path "*/Recalibrated/*.bam" |while read -r BAM
+  find "${ANALYSISDIR}/results/preprocessing/markduplicates/" -path "*/*.bam" -o -path "*/*.cram" | while read -r ALIGNMENTFILE
   do
-    BAMDIR="$(dirname "${BAM}")"
-    BAMNAME="$(basename "${BAM}")"
+    ALIGNMENTDIR="$(dirname "${ALIGNMENTFILE}")"
+    ALIGNMENTNAME="$(basename "${ALIGNMENTFILE}")"
 
     # construct the path for the metrics file
-    HSDIR="${BAMDIR/Preprocessing/Reports}"
-    HSDIR="${HSDIR/Recalibrated/HsMetrics}"
-    HSFILE="${BAMNAME/.bam/.hs_metrics}"
+    HSDIR="${ALNDIR/preprocessing/reports}"
+    HSDIR="${HSDIR/markduplicates/HsMetrics}"
+    EXTENSION="${ALIGNMENTNAME##*.}"
+    HSFILE="${ALIGNMENTNAME%.${EXTENSION}}.hs_metrics"
 
     mkdir -p "$HSDIR"
-    sbatch -J "${HSFILE}" -o "${HSDIR}/${HSFILE}.out" "$0" "$BAITS" "$TARGETS" "$ANALYSISDIR" "$BAM" "${HSDIR}/${HSFILE}"
+    sbatch -J "${HSFILE}" -o "${HSDIR}/${HSFILE}.out" "$0" "$BAITS" "$TARGETS" "$ANALYSISDIR" "$ALIGNMENTFILE" "${HSDIR}/${HSFILE}"
   done
 
 elif [[ $# -eq 5 ]]
 then
 
   # This is the functionality when the script is run as a sbatch job and launched with 5 parameters
 
-  singularity exec $CONTAINER gatk --java-options -Xmx28g CollectHsMetrics --INPUT=$INBAM --OUTPUT=$METRICS --MAX_RECORDS_IN_RAM=50000000 --BAIT_INTERVALS=$BAITS --TARGET_INTERVALS=$TARGETS
+  singularity exec $CONTAINER gatk --java-options -Xmx28g CollectHsMetrics \
+    --INPUT $INFILE \
+    --OUTPUT $METRICS \
+    --MAX_RECORDS_IN_RAM 50000000 \
+    --BAIT_INTERVALS $BAITS \
+    --TARGET_INTERVALS $TARGETS \
+    --REFERENCE_SEQUENCE $REF
```

Comment:
Just double checking, I guess there are no downsides with using a reference for BAM files?

Reply:
Coolio!